src/HasMedia/HasMediaTrait.php

```diff
@@ -315,6 +315,38 @@ public function clearMediaCollection(string $collectionName = 'default')
     }
 
     /**
+     * Remove all media in the given collection except some.
+     *
+     * @param string $collectionName
+     * @param \Spatie\MediaLibrary\Media[]|\Illuminate\Support\Collection $excludedMedia
+     *
+     * @return $this
+     */
+    public function clearMediaCollectionExcept(string $collectionName = 'default', $excludedMedia = [])
+    {
+        $excludedMedia = collect($excludedMedia);
+
+        if ($excludedMedia->isEmpty()) {
+            return $this->clearMediaCollection($collectionName);
+        }
+
+        $this->getMedia($collectionName)
+            ->reject(function (Media $media) use ($excludedMedia) {
+                return $excludedMedia->where('id', $media->id)->count();
+            })
+            ->each(function (Media $media) {
+                app(Filesystem::class)->removeFiles($media);
```

Comment:
Is it necessary to remove files manually here? I'm think the `MediaObserver` should take care of that.